cobaya/theories/classy/classy.py

```diff
@@ -298,10 +311,40 @@ def must_provide(self, **requirements):
     def add_z_for_matter_power(self, z):
         if getattr(self, "z_for_matter_power", None) is None:
             self.z_for_matter_power = np.empty(0)
-        self.z_for_matter_power = np.flip(np.sort(np.unique(np.concatenate(
-            [self.z_for_matter_power, np.atleast_1d(z)]))), axis=0)
+        self.z_for_matter_power = np.flip(combine_1d(z, self.z_for_matter_power))
         self.extra_args["z_pk"] = " ".join(["%g" % zi for zi in self.z_for_matter_power])
 
+    def set_collector_with_z_pool(self, k, zs, method, args=[], args_names=[], kwargs={},
+                                  arg_array=None, post=None):
+        """
+        Creates a collector for a z-dependent quantity, keeping track of the pool of z's.
+
+        If ``z`` is an arg, i.e. it is in ``args_names``, then omit it in the ``args``,
+        e.g. ``args_names=["a", "z", "b"]`` should be passed together with
+        ``args=[a_value, b_value]``.
+        """
+        if k in self.collectors:
+            z_pool = self.collectors[k].z_pool
+            z_pool.update(zs)
+        else:
+            z_pool = Pool1D(zs)
+        # Insert z as arg or kwarg
+        if "z" in kwargs:
+            kwargs = deepcopy(kwargs)
+            kwargs["z"] = z_pool.values
+        elif "z" in args_names:
+            args = deepcopy(args)
+            i_z = args_names.index("z")
+            args = args[:i_z] + [z_pool.values] + args[i_z:]
+        else:
+            raise LoggedError(
+                self.logger,
```

Comment:
self.log

Reply:
Sure, didn't notice that one, even though the linter complains. At some point, I'd like to change all the `log` attributes to `logger`, which is more precise. And maybe the `raise LoggedError` to a method?

Reply:
A method would be nice, but I think complicates the stack dump unless you hack it somehow